docs/stm32/general.rst

```diff
@@ -0,0 +1,70 @@
+.. _stm32_general:
+
+General Information about the STM32 port
+========================================
+
+STM32 is a family of microcontroller ICs based on 32-bit RISC architecture
+such as Cortex-M33F, Cortex-M7F, Cortex-M4F, Cortex-M3, Cortex-M0+, and
+Cortex-M0 cores.
+
+The STM microcontroller boards provides the facility of accessing the memory
+directly, to perform any read or write function.
+
+It exposes three objects::
+
+* stm.mem8 : access 8 bits of memory
+
+* stm.mem16 : access 16 bits of memory
+
+* stm.mem32 : access a 32 bit word memory
+
+They can be used in combination with the offset constants to read, write and
+control registers of the MCU hardware peripherals and all other areas of MCU's
+memory.
+
+Example:-
+
+..code-block:: python3
+        
+        # set PA2 high
+        stm.mem32[stm.GPIOA + stm.GPIO_BSRR] = 1 << 2
+
+        # read PA3
+        value = (stm.mem32[stm.GPIOA + stm.GPIO_IDR] >> 3) & 1
+        
+There are some constants for registers which are generated from a file after reading the
+alternate function table for the microcontroller unit that is being compiled for. Examples 
+of some constants include:-
+* GPIOA
+* GPIOB
+* GPIOC
+* GPIOD
+* GPIOE
+* GPIOF
+* GPIOG
+* GPIOH
+* GPIOI
+* GPIOJ
+* GPIOK
+* GPIO_AFR0
+* GPIO_AFR1
+* GPIO_BSRR
+* GPIO_BSRRH
+* GPIO_BSRRL
+* GPIO_IDR
+* GPIO_LCKR
+* GPIO_MODER
+* GPIO_ODR
+* GPIO_OSPEEDR
+* GPIO_OTYPER
+* GPIO_PUPDR
+    
+We also have the following functions:-
```

Comment:
These 3 functions are only available on STM32WB55 MCUs (so please add a note that says this).